### PR TITLE
Feature/EICNET-834: Implement the accept request response

### PR DESCRIPTION
**Pull request by korhankale**

### CODE REVIEW

File: lib/modules/eic_flags/src/Service/DeleteRequestHandler.php [hunk 1]
```diff
@@ -38,4 +66,124 @@ public function getSupportedEntityTypes() {
     ];
   }
 
+  /**
+   * @param \Drupal\group\Entity\GroupInterface $group
+   */
+  private function deleteGroup(GroupInterface $group) {
+    $related_stories = $group->get('field_related_news_stories')
```
Comment (lramarojaona) [inline lib/modules/eic_flags/src/Service/DeleteRequestHandler.php:73]: We should not delete related stories and news, as they only exist on a global level, they never belong to a group.

File: lib/modules/eic_flags/src/Service/DeleteRequestHandler.php [hunk 2]
```diff
@@ -38,4 +66,124 @@ public function getSupportedEntityTypes() {
...
+    // Retrieve group content entities linked to the group
+    $group_contents = $group->getContent();
+    $batch_builder = (new BatchBuilder())
+      ->setFinishCallback(
+        [
+          DeleteRequestHandler::class,
+          'deleteGroupContentFinished',
+        ]
+      )
+      ->setTitle($this->t('User updates'));
```
Comment (lramarojaona) [inline lib/modules/eic_flags/src/Service/DeleteRequestHandler.php:88]: User updates seem a bit confusing, can we have a more appropriate title here?
Such as "Deleting group content"?

**Review by lramarojaona — CHANGES_REQUESTED** [review]

File: lib/modules/eic_flags/src/Service/DeleteRequestHandler.php [hunk 3]
```diff
@@ -38,4 +66,118 @@ public function getSupportedEntityTypes() {
...
+        return;
+      }
+
+      if ($target_entity instanceof ContentEntityInterface) {
+        $target_entity->delete();
+      }
+
+    } catch (\Exception $exception) {
+      $context['results']['errors'][] = t(
+        'Something went wrong during content removal @error',
```
Comment (lramarojaona) [inline lib/modules/eic_flags/src/Service/DeleteRequestHandler.php:132]: Can we use $this->t() instead?

Comment (korhankale) [reply lib/modules/eic_flags/src/Service/DeleteRequestHandler.php:132]: Actually it is a static function so the answer is no without any negociation possible 😅

**Review by lramarojaona — APPROVED** [review]

**Pull request merged**